Add definition, support content, links to README (#101)
alphagov/govuk-country-and-territory-autocomplete: README.md

# GOV.UK country and territory autocomplete - what it is and how to use it

> The country and territory autocomplete is a component that helps users choose countries and territories in answers. You should only use this component for services where users have to provide their country or territory.

Reviewer: ```suggestion
The country and territory autocomplete is a component that helps users choose countries and territories in answers. You should only use this component for services where users have to provide their country or territory.

It includes complete functionality to make it faster and easier for users to find a location. For example, to select their country of birth or their current location.
 ```
 
 ... and then delete paragraph by line 7.

Author: That sounds like a good idea, thanks - will add in.

For other types of service, you should use the accessible autocomplete. To find information about this component, go to the [accessible-autocomplete README](https://github.com/alphagov/accessible-autocomplete/blob/main/README.md).

> The GOV.UK country and territory autocomplete  is a simple UI widget that allows end users to find and select a location. It includes auto-complete functionality to make it faster and easier for users to find a location. For example, to select their country of birth or their current location.

Author: Is this content worth somehow combining with the first sentence on this page? (Instead of having 2 similar pieces of content separated by line 5's content re the accessible-autocomplete.)

Reviewer: Perhaps move the 2nd sentence up after the 1st paragraph? 'Auto-complete' might not be needed (and spelled inconsistently) there.

Author: That sounds like a good idea - thanks. Will add in this change.

Author: Thanks - done!

Reviewer 2: `For other types of question` maybe? the type of service doesn't have a bearing on whether you'd use this component.

# GOV.UK country and territory autocomplete - what it is and how to use it

The country and territory autocomplete is a component that helps users choose countries and territories in answers. You should only use this component for services where users have to provide their country or territory.

This component includes complete functionality to make it faster and easier for users to find a location. For example, to select their country of birth or their current location.

> For other types of service, you should use the accessible autocomplete. To find information about this component, go to the [accessible-autocomplete README](https://github.com/alphagov/accessible-autocomplete/blob/main/README.md).

Author: ```suggestion
For other types of question, you should use the accessible autocomplete. To find information about this component, go to the [accessible-autocomplete README](https://github.com/alphagov/accessible-autocomplete/blob/main/README.md).
```

# GOV.UK country and territory autocomplete - what it is and how to use it

> The country and territory autocomplete is a component that helps users choose countries and territories in answers. You should only use this component for services where users have to provide their country or territory.

Author: First sentence has same wording as the [update for accessible-autocomplete](https://github.com/alphagov/accessible-autocomplete/pull/525/files):

> The [type-of-autocomplete] is a component that helps users choose countries and territories in answers.

Should we change the accessible-autocomplete wording to something like "The accessible autocomplete is a component that helps users choose answers from a list we provide"?

asdada

For other types of service, you should use the accessible autocomplete. To find information about this component, go to the [accessible-autocomplete README](https://github.com/alphagov/accessible-autocomplete/blob/main/README.md).

[![Build Status](https://travis-ci.com/alphagov/govuk-country-and-territory-autocomplete.svg?branch=main)](https://travis-ci.org/alphagov/govuk-country-and-territory-autocomplete)

> [![Greenkeeper badge](https://badges.greenkeeper.io/alphagov/govuk-country-and-territory-autocomplete.svg)](https://greenkeeper.io/)

Reviewer 3: These can be removed (if they haven't been already), as they no longer work. You might need to rebase your PR?

Author: Thanks, will check and, if they're there, remove.

> The GOV.UK country and territory autocomplete is a simple UI widget that allows end users to find and select a location. It includes auto-complete functionality to make it faster and easier for users to find a location. For example, to select their country of birth or their current location.

Author: Is this line worth combining with the first sentence on this page? (Instead of having 2 similar pieces of content separated by line 5's content re the accessible-autocomplete.)

Author: @calvin-lau-sig7

[Try out the example.](https://alphagov.github.io/govuk-country-and-territory-autocomplete/examples/)

---

## Support

The GOV.UK Design System team maintains the country and territory autocomplete. However, we’re only able to put in minimal work to support it.

> [Read about our plans to maintain this component](LINK-TO-PINNED-RESEARCH-ISSUE).

Author: ```suggestion
[Read about our plans to maintain this component](https://github.com/alphagov/govuk-country-and-territory-autocomplete/issues/106).
```

> [Read more about the types of support we can provide](LINK-TO-PINNED-SUPPORT-ISSUE).

Author: ```suggestion
[Read more about the types of support we can provide](https://github.com/alphagov/govuk-country-and-territory-autocomplete/issues/91).
```

[Try out the example.](https://alphagov.github.io/govuk-country-and-territory-autocomplete/examples/)

---

## Support

The GOV.UK Design System team maintains the country and territory autocomplete. However, we’re only able to put in minimal work to support it.

[Read about our plans to maintain this component.](LINK-TO-PINNED-RESEARCH-ISSUE)

> We provide support for:

Author: Might not want to use 'provide' here, if it implies we can only help users to fix these things, rather than fixing them ourselves. [See comment on other autocomplete PR for more context](https://github.com/alphagov/accessible-autocomplete/pull/525/files#r785790989).

- accessibility issues that block usage
- security issues
- broken functionality that blocks usage for most users
- dependency updates to keep tests working so we can make changes safely
- documentation updates

We cannot currently provide support for:

- adding new features
- smaller functionality issues that do not block usage
- issues caused by users customising the country and territory autocomplete
- issues with browsers and assistive technologies not mentioned in the [GOV.UK Service Manual](https://www.gov.uk/service-manual/technology/designing-for-different-browsers-and-devices)
- complex issues with React

> [Read more about the types of support we can provide.](LINK-TO-PINNED-SUPPORT-ISSUE)

Author: ```suggestion
[Read more about the types of support we can provide](LINK-TO-PINNED-SUPPORT-ISSUE).
```

[Try out the example.](https://alphagov.github.io/govuk-country-and-territory-autocomplete/examples/)

---

## Support

> The GOV.UK Design System team maintains the country and territory autocomplete. However, we’re only able to put in minimal work to maintain and support it.

Author: ```suggestion
The GOV.UK Design System team maintains the country and territory autocomplete. However, we’re only able to put in minimal work to support it.
```

> [Read about our plans to maintain this component.](LINK-TO-PINNED-RESEARCH-ISSUE)

Reviewer 3: I think the style guide says we should put full stops outside of links?

```suggestion
[Read about our plans to maintain this component](LINK-TO-PINNED-RESEARCH-ISSUE).
```

Reviewer 3: Weirdly it's in the [Design System style guide](https://design-system.service.gov.uk/styles/typography/#links), but not the [GOV.UK style guide](https://www.gov.uk/guidance/content-design/links) 🤔 Is there a disagreement here, or just one of those weird overlaps?

Reviewer 2: I think where the link is the entire sentence, you can link it - especially if separate from other paragraphs.

Author: To stay consistent with what we've done elsewhere in Design System content, might make sense to put full stops outside the links. For example, the way we've displayed the link text in the What's New content on our [homepage](https://design-system.service.gov.uk/).

Worth looping @calvin-lau-sig7 into this convo...

Reviewer: The guidance on the [Typography](https://design-system.service.gov.uk/styles/typography/#links) page? I'd say it covers 'services' but not 'flat content'... I know the boundary can be fuzzy sometimes!

Author: @calvin-lau-sig7 Yeah, exactly. Was thinking the README falls under 'flat content'.

Reviewer: Yeah, readme as 'flat content' sounds right, up to you of course!

I think a full sentence is fine to include the full stop — and that's how I'd interpret the guidance in Typography. As in, 'at the end' means the same as 'the last part.'

> If your link is at the end of a sentence or paragraph, make sure that the linked text does not include the full stop.

Not sure there's a screen reader consideration or something I don't know about? Otherwise I might go and tweak the guidance in 'Typography.'

Author: @calvin-lau-sig7 I've been looking at our READMEs and release notes for entire sentences that are links. Since this content is also on GitHub, it seems like a fair point of comparison. Looks like our custom is, just like Design System guidance and Frontend docs, to leave the full stop unlinked. So maybe we can stay consistent with that and go with @lfdebrux's suggested changes.
